sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/OperatorStateMetadata.scala

```diff
@@ -60,19 +81,65 @@ case class OperatorStateMetadataV1(
   override def version: Int = 1
 }
 
-object OperatorStateMetadataUtils {
+case class OperatorStateMetadataV2(
+    operatorInfo: OperatorInfoV1,
+    stateStoreInfo: Array[StateStoreMetadataV2],
+    operatorPropertiesJson: String) extends OperatorStateMetadata {
+  override def version: Int = 2
+}
+
+object OperatorStateMetadataUtils extends Logging {
+
+  sealed trait OperatorStateMetadataReader {
+    def version: Int
+
```

Comment:
nit: shall we be consistent? see below trait - no empty line between two or both empty line